openhgnn/sampler/MAGNN_sampler.py

```diff
@@ -0,0 +1,195 @@
+    ----------
+    g : dgl.DGLGraph
+        the dgl graph from which we sample subgraphs.
+    mask : list or numpy.ndarray
+        the train/test/test nodes mask, which should be like train mask = [0, 1, 0, 0, 1] indicating that the
+        1th, 4th nodes are training nodes of g. Note that the mask indicates which part of nodes in g will be sampled
+        by MAGNN sampler. e.g. if it's the test mask, the sample will be employed on testing nodes of g to sample
+        subgraphs for testing phase. This param is useful when the graph is too large to be directly validate on.
+    n_layers : int
+        the number of layers of each subgraph.
+    category : any
+        the class of seed_nodes.
+    metapath_list : list
+        the type of all metapaths in graph g. e.g. ['MAM', 'MDM', ...]
+    num_samples : int
+        the maximal number of metapath instances we'd like to sample for each node. i.e. for each node,
+        if the number of metapath instances is more than num_samples, we sample num_samples instances randomly based on
+        a specific calculated probability. Otherwise, we sample all instances.
+    dataset_name : str
+        the name of the dataset.
+        Default : 'dblp4MAGNN'
+
+    '''
+    def __init__(self, g, mask, n_layers, category, metapath_list,
+                 num_samples, dataset_name='dblp4MAGNN'):
+        self.g = g
+        self.mask = mask
+        self.dataset_name = dataset_name
+        self.metapath_list = metapath_list
+        self.n_layers = n_layers
+        self.category = category
+        self.num_samples = num_samples
+        self.mp_inst = mp_instance_sampler(g, self.metapath_list, self.dataset_name)
+
+    def __getitem__(self, idx):
+        '''
+        Parameters
+        ----------
+        idx : int
+            The node ids of ntype as self.category, that is, the original seed_nodes is {self.category: idx}.
+
+        Returns
+        -------
+        dict
+            The nids sampled by the MAGNN_sampler from the original graph. The nids are of original version.
+        dict
+            The metapath instances corresponding to the sampled graph. The nids in mp instances are of original version.
+        DGLGraph
+            The original heterogeneous subgraph.
+
+        Notes
+        -----
+        Sampling neighbors based on seed_nodes and metapath instances mp_inst[i],
+        we only need to sample mp_inst from all metapath instances with seed_nodes as mp_inst[i][0] and
+        sample neighbors as mp_inst[i][1, 2, ...]. Here we consider mp_inst[i][0] as dst_nodes.
+
+        But the sampled mp_inst above does not contain all the mp_inst of the sampled subgraph.
+        For example, if seed_nodes contain M1 and we sample neighbors as A1, M2 based on metapath instance M1-A1-M2,
+        we'd lose metapath instance like M2-A1-M1 because M2 is not one of the seed_nodes while M2-A1-M1 absolutely
+        one of the metapath instances of the sampled subgraph.
+
```

Comment:
Is it saying that the sampled mp_inst does not necessarily cover all the possible metapaths?  Probably not important since you already specify what to sample in `metapath_list`.

Reply:
These comments are a bit confusing and actually they are useless for using this sampler...I'll delete them.